rotkehlchen/externalapis/etherscan.py

```diff
@@ -285,11 +339,34 @@ def get_transactions(
             for entry in result:
                 gevent.sleep(0)
                 try:
+                    # Handle genesis block transactions
+                    if entry['hash'].startswith('GENESIS'):
+                        if self.db is None:
+                            raise DeserializationError('DB not initialized')
+                        trace_id = self._latest_trace_id(self.db)
+                    else:
+                        trace_id = None
+
                     tx = deserialize_ethereum_transaction(  # type: ignore
                         data=entry,
                         internal=is_internal,
                         ethereum=None,
+                        trace_id=trace_id,
                     )
+
+                    if entry['hash'].startswith('GENESIS'):
+                        # Check for null self.db twice as mypyi doesn't check that
+                        # self.db check was already done under same condition
+                        if self.db is None:
+                            raise DeserializationError('DB not initialized')
+
+                        genesis_txn = self._get_or_create_genesis_transaction(self.db, account)
```

Comment:
`_get_or_create_genesis_transaction` returns a usual ethereum transactions but here you are using it as an internal transaction. Sounds incorrect, doesn't it?

Reply:
Oh, I read the code incorrectly, apologies

Reply:
But [this comment](https://github.com/rotki/rotki/pull/4475/files#r897764620) is kinda related and is important in my opinion. What do you think about it?